extensions/cache/runtime/src/main/java/io/quarkus/cache/build/CacheBuildRecorder.java

```diff
@@ -0,0 +1,34 @@
+package io.quarkus.cache.build;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.cache.augmented.interceptor.CacheRepository;
+import io.quarkus.cache.augmented.interceptor.CaffeineCache;
+import io.quarkus.runtime.annotations.Recorder;
+
+/**
+ * @author Gwenneg Lepage
+ */
+@Recorder
+public class CacheBuildRecorder {
```

Comment:
I'm not sure this is absolutely necessary, but it seemed great to build cache instances at build time using bytecode generation.